src/main/java/org/lwes/emitter/DatagramSocketEventEmitter.java

```diff
@@ -0,0 +1,278 @@
+   * @return the ESF file path
+   */
+  public String getESFFilePath() {
+    if (getFactory() != null) {
+      return getFactory().getESFFilePath();
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Sets an InputStream to be used for event validation.
+   * @param esfInputStream an InputStream used for event validation
+   */
+  public void setESFInputStream(InputStream esfInputStream) {
+    if (getFactory() != null) {
+      getFactory().setESFInputStream(esfInputStream);
+    }
+  }
+
+  /**
+   * Gets the InputStream being used for event validation.
+   * @return the InputStream of the ESF validator
+   */
+  public InputStream getESFInputStream() {
+    if (getFactory() != null) {
+      return getFactory().getESFInputStream();
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Creates the underlying datagram socket.
+   * @throws IOException if an I/O error occurs during socket creation.
+   */
+  protected abstract void createSocket() throws IOException;
+
+  /**
+   * Initializes the emitter.
+   * @throws IOException if an I/O error occurs during initialization.
+   */
+  @Override
+  public void initialize() throws IOException {
+    createSocket();
+    super.initialize();
+  }
+
+  /**
+   * @throws IOException if an I/O error occurs on shutdown.
+   */
+  public void shutdown() throws IOException {
+    // close the socket AFTER calling super shutdown since that is trying to
+    // send a shutdown message.
+    super.shutdown();
+    if (socket != null) {
+      socket.close();
+    }
+  }
+
+  /**
+   * Creates a new event named <tt>eventName</tt>.
+   * @param eventName the name of the event to be created
+   * @return a new Event
+   * @exception EventSystemException if there is a problem creating the event
+   */
+  public Event createEvent(String eventName) throws EventSystemException {
+    return createEvent(eventName, true);
+  }
+
+  /**
+   * Creates a new event named <tt>eventName</tt>.
+   * @param eventName the name of the event to be created
+   * @param validate whether or not to validate the event against the EventTemplateDB
+   * @return a new Event
+   * @exception EventSystemException if there is a problem creating the event
+   */
+  public Event createEvent(String eventName, boolean validate) throws EventSystemException {
+    if (getFactory() != null) {
+      return getFactory().createEvent(eventName, validate);
+    } else {
+      throw new EventSystemException("EventFactory not initialized");
+    }
+  }
+
+  /**
+   * Emits the event to the network.
+   *
+   * @param event the event to emit
+   * @exception IOException throws an IOException is there is a network error.
+   * @throws EventSystemException if unable to serialize the event
+   * @return number of bytes emitted
+   */
+  public int emit(Event event) throws IOException, EventSystemException {
+    byte[] msg = event.serialize();
+    int bytesEmitted = 0;
+    synchronized (lock) {
```

Comment:
I know this bit was in the original code, but do we really need to synchronize here?  emit basically just calls socket.send, which I think is threadsafe.  collectStatistics does stuff that doesn't appear threadsafe, so maybe there's little to gain in calling emit outside of a synchronized block.  one could probably also make collectStatistics threadsafe as well, but I'm just venturing further away from the intent of this PR.